salvo/api/image.proto

```diff
@@ -34,4 +34,11 @@ message DockerImages {
   // defined and benchmark the discovered image.
   // eg: [ "envoyproxy/envoy-dev:7da2adee8962c202999e96cb41e899deb34faf48",..]
   repeated string additional_envoy_images = 5;
+
+  // If this field is true and additional_envoy_images is not defined, we will
+  // only test envoy image which is defined in envoy_image field. It's useful
+  // for the scenario that someone want to test different configuration with 
+  // specific envoy image. test_single_image and additional_envoy_images are
+  // mutually exclusive and cannot be defined at the same time.
+  bool test_single_image = 6;
```

Comment:
Sorry, this didn't occur to me before. Since the fields are mutually exclusive, would it be better to define them as a oneof proto field?

https://developers.google.com/protocol-buffers/docs/proto3#oneof

Applies to the change in `source.proto` as well.

Reply:
It's a nice suggestion and I've tried to define them as a `oneof` proto like:
```
  oneof test_mode {
    // Specify a set of additional hashes for comparing benchmark results.
    // This field is optional and if unspecified, we will default to using
    // one commit prior to the specified "commit_hash" field. For each
    // hash specfied we will run a benchmark using the Envoy binary built
    // from that source commit
    repeated string additional_hashes = 7;
    
    // If this field is true and additional_hashes is not defined, we will
    // only test envoy commit which is defined in commit_hash field. It's useful
    // for the scenario that someone want to test different configuration with 
    // specific envoy commit. test_single_commit and additional_hashes are
    // mutually exclusive and cannot be defined at the same time
    bool test_single_commit = 8;
  }
```

But the fileds in `oneof` cannot have label `repeated`: 
```
api/source.proto:52:5: Fields in oneofs must not have labels (required / optional / repeated).
```
If we delete repeated label, then it means `additional_hashes` cannot defined as a list, such as `additional_hashes: [02dff6e0235e2f497fb57eb0b7c07cc091c538df, b435d3a3baa39f1a15cd68625da085cfa16ae957]`.

Reply:
That is a good point, sorry for missing that. We typically go around that by wrapping the repeated field in a new message, but we shouldn't do that here, since that would be a breaking API change.

Let's proceed with what we have now. Thank you for exploring this option.